src/AutoMapper/MapperConfiguration.cs

```diff
@@ -108,6 +111,16 @@ public TypeMap ResolveTypeMap(Type sourceType, Type destinationType)
         public TypeMap ResolveTypeMap(TypePair typePair)
         {
             var typeMap = _typeMapPlanCache.GetOrAdd(typePair, _getTypeMap);
+
+            _typeMapPlanCache.AddOrUpdate(typePair, typeMap, (tp, tm) =>
+            {
+                if (tm?.Sealed == false)
+                    tm.Seal(_typeMapRegistry, this);
+                else if (typeMap?.Sealed != true)
+                    Task.Delay(1).Wait();

```

Comment:
I can't seal the newly generated type maps as they are being created with a circular loop.  It tries to resolve the other type map which then resolves itself and over and over.  This resolves the type map, and then seals it after it's added to cache like regular TypeMaps are.  And to break the inifinite seal loop, I have to know it's in the process of being sealed and not try to re-seal it.

The delay it to make the dynamic mapping multi-threaded test pass